aten/src/ATen/native/cuda/IndexKernel.cu

```diff
@@ -24,53 +25,23 @@ static constexpr int launch_bound2 = 4;
 
 static constexpr int launch_size_nd = 128;
 
-template <int Dims, typename T, typename IndexType>
-__device__ __forceinline__ IndexType indexToOffset(
-    const cuda::detail::TensorInfo<T, IndexType>& info,
+template <typename scalar_t, typename index_t, int S>
+__device__ __forceinline__ index_t indexToOffset(
```

Comment:
sorry it's late in the game for this, but is there any reason you are not using OffsetCalculator https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cuda/detail/OffsetCalculator.cuh for this? It is faster as it avoids integer division, and it will allow you to not create TensorInfo for arguments, as it acts on strides and sizes directly. It's not critical for this PR though. You can see an example of how it's used here https://github.com/pytorch/pytorch/blob/33b95c6bac5cf9c52fa646bf7335664b0556263d/aten/src/ATen/native/cuda/IndexKernel.cu#L132-L134, but you don't have to create offsetCalculator for all iter arguments, you can do it just for one, as you need here (see e.g. make_output_offset_calculator utility).

Reply:
I just pushed these changes. Interestingly enough, my benchmarks show that `indexToOffset` is more efficient than this implementation... This could be something to look at at some point